### ISSUE TITLE
Target 255 is out of bounds

### ISSUE THREAD

**Issue by BrsDincer** [Deci-AI/super-gradients#1443]
### 💡 Your Question

I got an error when I was working on the `colab` you defined and referenced in this "**https://docs.deci.ai/super-gradients/latest/documentation/source/Segmentation.html#load-the-model-from-modelzoo**" page.

I think there is a problem with the version. Installation of "**super_gradients==3.0.5**" is not happening on the kernel. Then I did a development with up-to-date "**super_gradients**" and didn't use the old version. But even if I follow the necessary instructions, I continue to get the error I mentioned in the title "Target 255 is out of bounds". This happens in the "_trainer_" section.

I am writing the code flow below:

**Training constant:**

```
train_parameters = {"max_epochs": 25,
                "lr_mode": "cosine",
                "initial_lr": 0.005,
                "optimizer": "SGD",
                "loss": "cross_entropy",
                "average_best_models": False,
                "metric_to_watch": "IoU",
                "greater_metric_to_watch_is_better": True,
                "train_metrics_list": [IoU(num_classes=10)],
                "valid_metrics_list": [IoU(num_classes=10)],
                "loss_logging_items_names": ["loss"],
                "phase_callbacks": [BinarySegmentationVisualizationCallback(phase=Phase.VALIDATION_BATCH_END,
                                                                            freq=1,
                                                                            last_img_idx_in_batch=4)],

                }
```

**Model PP LITE T SEG75:**

```
model_seg75 = models.get(model_name=Models.PP_LITE_T_SEG75,
                         arch_params={"use_aux_heads": False},
                         num_classes=10,
                         pretrained_weights="cityscapes")
```

**CustomDataset(Dataset)**:

```
CustomDataset(Dataset)
    def __init__(self,data_folder,split)->CLASSINIT:
        self.data_folder = data_folder
        self.split = split.lower()
        assert self.split in {'train', 'valid'}
        with open(os.path.join(data_folder, self.split + '.lst'), 'r') as f:
            data_lines = f.readlines()
        self.samples_fn = [line.strip().split(" ") for line in data_lines]
        self.transforms = transforms.Compose([transforms.ToTensor()])
    def __getitem__(self, i)->CLASSINIT:
        image = Image.open(self.samples_fn[i][0]).convert('RGB')
        label = Image.open(self.samples_fn[i][1])
        image = image.resize((768,1536))
        label = label.resize((768,1536))
        image_tensor = self.transforms(image)
        label_tensor = torch.from_numpy(np.array(label)).long()
        return image_tensor, label_tensor
    def __len__(self)->CLASSINIT:
        return len(self.samples_fn)
```

**Train and validation sets:**

```
train_dataset = CustomDataset(main_data_path, split="train")
val_dataset = CustomDataset(main_data_path, split="valid")
```

Loaders:

```
train_loader_data = DataLoader(train_dataset,
                               batch_size=4,
                               shuffle=True,
                               num_workers=2)

validation_loader_data = DataLoader(val_dataset,
                                    batch_size=4,
                                    shuffle=True,
                                    num_workers=2)
```

**Trainer:**

```
trainer_main = Trainer(experiment_name=exp_name,ckpt_root_dir=chk_path)

trainer_main.train(model=model_seg75,
                   training_params=train_parameters,
                   train_loader=train_loader_data,
                   valid_loader=validation_loader_data)
```

**_TORCH Version_**: 2.0.1+cu118
**_CUDA_** is running actively.

There is no problem in the file pre-process stages. But none of the segmentation approaches with custom data I tried in the current version work. I would appreciate if you could send me the solution of this or a sample `colab` code that I can perform segmentation up to date.

**Comment by BloodAxe** [comment]
Looks like [Colab notebook](https://colab.research.google.com/drive/1d7cU0NsUj7jnOF1YSap_DH9r79G3-Cr4?usp=sharing#scrollTo=cuuKelmO9QoB) should be updated to reflect the breaking changes that happened between 3.0.5 and 3.2.0.

**Comment by shaydeci** [comment]
Notebooks have been updated and the problem doesnt seem to persist.
Please re-open in case it does for you @BrsDincer .
